### DISCUSSION TITLE
New command in addition to `project deploy preview` and `project retrieve preview`?

### DISCUSSION THREAD

**Discussion by k-capehart** [forcedotcom/cli#2778] [Ideas]
I find myself often running both `sf project deploy preview` and `sf project retrieve preview` one after the other to get an idea of any pending changes originating in either my org or my local files. 

I understand the benefit of having this split into 2 commands, but what are thoughts regarding a new separate command that essentially outputs the result of both `deploy preview` and `retrieve preview`? And any thoughts on what the command would be called?

**Comment by cristiand391** [comment]
Related (see 1st use-case listed):
https://github.com/forcedotcom/cli/discussions/2040

I could imagine this also more like an IDE feature where you could have file watcher on your project to show that will be deployed when saving changes and something similar to ` project retrieve preview` that runs every few X minutes.

**Reply by k-capehart** [reply]
Thanks for linking this. One thing I'll add though is that while the solution outlined in #2040 is useful, it might be overkill for my specific use case. I'd want a similar output as the deploy/retrieve commands but without having to run both. Ideally it would output something like this:

```
Metadata to be retrieved
Type        Fullname
-----       ----------
type1       name1

Metadata to be deployed
Type        Fullname
-----       ----------
type2       name2

Metadata conflicts
Type        Fullname
-----       ----------
type3       name3
```

I'm not making any assumptions as to whether the org or the project is the "target". Instead I just want to see the differences so I can pick and choose what to deploy and what to retrieve.
